src/Microsoft.Diagnostics.Runtime/src/Common/ClrArray.cs

```
@@ -127,6 +117,139 @@ public int GetLength(int dimension)
+                if (IsMultiDimensional)
+                {
+                    valueOffset = indices[0] - GetMultiDimensionalBound(1);
+                    if ((uint)valueOffset >= GetMultiDimensionalBound(0))
+                        throw new ArgumentOutOfRangeException(nameof(indices));
+
+                    dataByteOffset += 2 * sizeof(int);
+                }
+                else
+                {
+                    valueOffset = indices[0];
+                    if ((uint)valueOffset >= Length)
+                        throw new ArgumentOutOfRangeException(nameof(indices));
+                }
+            }
+            else
+            {
+                for (int dimension = 0; dimension < rank; dimension++)
+                {
+                    int currentValueOffset = indices[dimension] - GetMultiDimensionalBound(rank + dimension);
+                    if ((uint)currentValueOffset >= GetMultiDimensionalBound(dimension))
+                        throw new ArgumentOutOfRangeException(nameof(indices));
+
+                    valueOffset *= GetMultiDimensionalBound(dimension);
+                    valueOffset += currentValueOffset;
+                }
+
+                dataByteOffset += 2 * sizeof(int) * rank;
+            }
+
+            int valueByteOffset = dataByteOffset + valueOffset * Type.ComponentSize;
+            return Type.ClrObjectHelpers.DataReader.Read<T>(Address + (ulong)valueByteOffset);
+        }
+
+        public ClrObject GetObjectValue(int index)
+        {
+            ulong address = GetValue<UIntPtr>(index).ToUInt64();
+            if (!Type.IsObjectReference)
+                throw new InvalidOperationException();
+
+            return Type.Heap.GetObject(address);
+        }
+
+        public ClrObject GetObjectValue(params int[] indices)
+        {
+            ulong address = GetValue<UIntPtr>(indices).ToUInt64();
+            if (!Type.IsObjectReference)
+                throw new InvalidOperationException();
```

`throw new InvalidOperationException();` maybe this should be `throw new InvalidOperationException($"{Type} does not contain object references.");` or something similarly more descriptive.